File: majorProject/src/main/java/com/aocc/majorproject/VersionOverlay.java
package com.aocc.majorproject;

import android.graphics.Color;
import android.graphics.Paint;

import com.aocc.framework.Graphics;

public final class VersionOverlay {

    private static final int VERSION_X = 1265;
    private static final int VERSION_Y = 710;

    private VersionOverlay() {
    }

    public static void paint(Graphics g, Paint paint) {
        if (Assets.plain != null) {
            paint.setTypeface(Assets.plain);
        }
        paint.setTextAlign(Paint.Align.RIGHT);
        paint.setTextSize(18);
        g.drawString("v" + BuildConfig.VERSION_NAME, VERSION_X, VERSION_Y,
                Color.argb(170, 255, 255, 255), paint);
    }
}

File: majorProject/src/main/kotlin/com/aocc/majorproject/VersionOverlay.kt
package com.aocc.majorproject

import android.graphics.Color
import android.graphics.Paint
import com.aocc.framework.Graphics

object VersionOverlay {

    private const val VERSION_X = 1265
    private const val VERSION_Y = 710
    private const val VERSION_TEXT_SIZE = 18f

    private val VERSION_PAINT = Paint(Paint.ANTI_ALIAS_FLAG)

    @JvmStatic
    fun paint(g: Graphics) {
        if (Assets.plain != null) {
            VERSION_PAINT.typeface = Assets.plain
        }
        VERSION_PAINT.textAlign = Paint.Align.RIGHT
        VERSION_PAINT.textSize = VERSION_TEXT_SIZE
        VERSION_PAINT.color = Color.argb(170, 255, 255, 255)
        g.drawString(
            "v${BuildConfig.VERSION_NAME}",
            VERSION_X,
            VERSION_Y,
            VERSION_PAINT.color,
            VERSION_PAINT
        )
    }

    /** @deprecated Use [paint] — version label uses its own paint. */
    @JvmStatic
    @Deprecated("Use paint(Graphics)", ReplaceWith("paint(g)"))
    fun paint(g: Graphics, paint: Paint) {
        paint(g)
    }
}
